app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -22,11 +22,19 @@ def openid_connect
 
       identifier_scheme = IdentifierScheme.find_by(name: auth.provider)
 
-      if current_user.nil?
-        # We need to register
-        if user.nil?
-          # Register and sign in
+      if current_user.nil? # if user is not signed in (They clicked the SSO sign in button)
+        if user.nil? # If an entry does not exist in the identifiers table for the chosen SSO account
           user = User.create_from_provider_data(auth)
+          if user.nil? # if a user was NOT created (a match was found for User.find_by(email: auth.info.email)
+            # Do not link SSO credentials for the signed out, existing user
+            flash[:alert] = _('The email you selected has not yet been linked to an existing account.<br>' \
+                              "Please sign in via the 'Sign in' button and navigate to the " \
+                              "'Edit Profile' section of DMP Assistant.<br>" \
+                              'From there you can link an email and enable single sign on access.<br>')
```

Comment:
We need to make this message simpler. Something along the lines of:

_The email provided has already been registered as an existing account._ Please sign in via the 'Sign in' button and navigate to the 'Edit Profile' section of DMP Assistant. From there you can link an email and enable single sign on access

Youll notice I only modified the first sentence. Maybe in can still be further improved ?

Reply:
I agree, it is a little complex. Maybe we should get Marcus to decide on the final result?

Reply:
How about the following?

```
flash[:alert] = _('You must be signed in to link the provided email.<br>' \
                              "Please sign in via the 'Sign in' button and navigate to the " \
                              "'Edit Profile' section of DMP Assistant.<br>" \
                              'From there you can link the provided email to enable single sign on access.<br>')
```

![Screenshot from 2024-09-17 09-08-49](https://github.com/user-attachments/assets/dd260480-30ab-4955-9da2-b31766b8292a)

Reply:
We should be clear that the email is already used as a system account. The user needs to sign in using that system account with that email, and then they can link the account.